examples/passport-oauth2-login/src/authentication-strategies/oauth2-strategy.ts

```diff
@@ -0,0 +1,104 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: example-passport-oauth2-login
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  asAuthStrategy,
+  AuthenticationStrategy,
+  UserProfileFactory,
+} from '@loopback/authentication';
+import {StrategyAdapter} from './strategy-adapter';
+import {Strategy, StrategyOption, Profile} from 'passport-facebook';
+import {bind, inject} from '@loopback/context';
+import {Request} from '@loopback/rest';
+import {UserProfile, securityId} from '@loopback/security';
+import {UserService, UserServiceBindings} from '../services';
+import {User} from '../models';
+import {asSpecEnhancer, OASEnhancer} from '@loopback/openapi-v3';
+
+const options: StrategyOption = {
+  clientID: '202843417684514',
+  clientSecret: 'ad4315e3453b012f6e4b8dd4bc1c0ae6',
+  callbackURL: '/auth/thirdparty/callback',
+};
+
+@bind(asAuthStrategy, asSpecEnhancer)
+export class Oauth2Authorization
+  implements AuthenticationStrategy, OASEnhancer {
+  name = 'Oauth2';
+  facebookStrategy: Strategy;
+  facebookAuthentication: StrategyAdapter<Profile>;
+
+  constructor(
+    @inject(UserServiceBindings.USER_SERVICE)
+    public userService: UserService,
+  ) {
+    this.facebookStrategy = new Strategy(options, this.verify.bind(this));
+    this.facebookAuthentication = new StrategyAdapter(
+      this.facebookStrategy,
+      'facebook',
+      mapFaceBookProfile,
+    );
+  }
+  modifySpec(
+    spec: import('@loopback/rest').OpenAPIObject,
```

Comment:
What is being done here exactly? What is being added to the app's open api spec? thx.

Reply:
this is not relevant, removed now.